modules/Bio/EnsEMBL/Compara/RunnableDB/GeneTrees/OrthologyStats.pm

```diff
@@ -68,19 +68,9 @@ sub fetch_input {
     my $genome_dbs   = $mlss->species_set->genome_dbs;
     my $gdb_id_1     = $genome_dbs->[0]->dbID;
     my $gdb_id_2     = $genome_dbs->[1]->dbID;
-    
-    # create map between (potential) component genome_db_ids and their principal genome_db_id
-    my $gdb_map;
-    foreach my $gdb ( @$genome_dbs ) {
-        my $comp_gdbs = $gdb->component_genome_dbs;
-        if ( defined $comp_gdbs->[0] ) {
-            foreach my $comp_gdb ( @$comp_gdbs ) {
-              $gdb_map->{$comp_gdb->dbID} = $gdb->dbID;
-            }
-        } else {
-            $gdb_map->{$gdb->dbID} = $gdb->dbID;
-        }
-    }
+
+    # create map between genome names and ids - excluding ids for component genomes
+    my %gdb_map = map { $_->name => $_->dbID } (grep {not $_->genome_component} @$genome_dbs);
```

Comment:
Can you explain what has changed to require this ?

Reply:
When I made this change, the file didn't have the `genome_db_id`, only the species name. It wasn't until a little later I discovered another analysis that required `genome_db_id` to be there too, so it was included then. Didn't seem worth changing this back as it's a lot more concise this way.

Reply:
I see. I was concerned because the way the polyploids are handled differs. TBH I can't tell how the entire PR is working on polyploids, but I guess we can see things live in production